packages/gitbook/src/components/SiteLayout/ClientContexts.tsx

```diff
@@ -24,7 +26,8 @@ export function ClientContexts(props: {
 
     return (
         <ThemeProvider nonce={nonce} attribute="class" enableSystem forcedTheme={forcedTheme}>
-            <LinkSettingsContext.Provider value={{ externalLinksTarget }}>
+            <LinkSettingsContext.Provider value={{ externalLinksTarget, contextId }}>
```

Comment:
You probably don't need it in the context if we can pass it directly as a prop to `RouterCacheClearer`